code/__DATASTRUCTURES/linked_lists.dm

```diff
@@ -0,0 +1,173 @@
+
+//Ok so it's technically a double linked list, bite me.
+
+/datum/linked_list
+	var/list/contents = list()
```

Comment:
to the best of our knowledge, byond lists are red-black trees combined with something else for associativity.